### DISCUSSION TITLE
MLFlow Pipelines vs SageMaker pipelines

### DISCUSSION THREAD

**Discussion by 99snowleopards** [mlflow/mlflow#6306] [General]
MLFlow pipelines looks very very promising! It'd be very useful to have a comparison of the current/planned features/functionality in MLFlow pipelines vs SageMaker pipelines, and maybe even KubeFlow (pipelines). 

Is the vision for MLFlow pipelines to eventually encourage teams that are currently using SageMaker pipelines to consider switching to MLFlow pipelines?

Thanks,

**Comment by mengxr** [comment]
MLflow Pipelines provide abstractions at higher level than SageMaker Pipelines and KubeFlow. With pre-defined pipeline templates, e.g., regression template, data scientists do not need to assemble an ML pipeline end-to-end to solve problems. They just need to configure some steps and provide code for the core logic (feature transformation, estimators, etc) to run the pipeline end-to-end.

**Comment by ahdbilal** [comment]
@99snowleopards if you did get the chance to use the experimental version of the pipeline, we would love to talk to you to get your feedback. if interested, you can use this [form](https://docs.google.com/forms/d/e/1FAIpQLSeYh6-u0qFn7AYv06Kiph09oZq2PW7Vx3Z_qVjFLE5mfMWQFA/viewform?usp=sf_link) to sign up and we will reach out to you via email. Thank you!

**Comment by 99snowleopards** [comment]
Hi @ahdbilal - I'd love to use the current version of the pipeline, but need a clearer understanding of how MLFlow pipelines would fit with either SageMaker pipelines or Kubeflow (pipelines) - the reply from @mengxr helps. I think it's important to a lot of companies to minimize the number of different tools/workflow libraries that are used to run their pipelines.

I'll look into this some more and fill out the form, thank you
